### DISCUSSION TITLE
Potential issue with strict_thresholding=True

### DISCUSSION THREAD

**Discussion by harrietgilmour** [tobac-project/tobac#314] [Q&A]
Hi all,

I’ve started trying out the multiple n_min_threshold (#208) and strict thresholding (#283) features in the new version release (1.5.0) and I’m getting some slightly strange results so I thought I’d ask on here to see if it’s something I’m doing wrong or if it’s a strange bug. 

I’m running tobac over a month of model data over South America to track MCSs. As a result, I wanted to make use of #208 to give a different area requirement to the deep convective core within the larger cloud shield. Here, I am using threshold = [240, 200] and n_min_threshold = [1975, 10]. This seems to work well and as expected. 

However, the issues arise when I also specify ‘strict_thresholding = True’ (#283). At the time, I was also testing the sensitivity of the cold core / deep convective core threshold (200 in above example). To do this, I reduced the value by 10 for each run. What was strange is that tobac detected features and tracks below the minimum value of the whole datatset (i.e. the minimum brightness temp value of the dataset is 179.43K, yet tobac still detected features as reaching the 170K threshold. This continues to occur as I reduce the cold core threshold even lower. 

Perhaps this screenshot below explains this more clearly:
![tobac_170K_issue](https://github.com/tobac-project/tobac/assets/90762936/7755a032-62dc-4e4f-8e9f-a62ef3d65bed)

You can see here that the minimum brightness temp value over the whole month is 179.43K. With strict_thresholding=False, there are no features that are detected as reaching the 170K threshold (as expected). However, with strict_thresholding=True, there are 2707 features detected below the 170K threshold. This shouldn’t occur as this is below the minimum value of the dataset. 

Similarly with the tracks, tobac detects 44 tracks when 0 should be detected (as we get when strict_thresholding=False). 

I then wanted to see where exactly tobac was finding these <170K tracks. They certainly aren’t where I’d expect the lowest brightness temperature values to be, even if 170K was reached and is making me even more confused!

![tobac_map](https://github.com/tobac-project/tobac/assets/90762936/716b87e5-145f-4c9d-a5e7-b4a74d26241c)

I should also add that this is my features set up (definitely using target=’minimum’):
![tobac_feature_setup](https://github.com/tobac-project/tobac/assets/90762936/c337fabc-f2bd-41c0-b36b-25b8a02ffdf0)

I’d love to hear that I’m just doing something wrong so if I am, please let me know! Otherwise, it may be some strange bug in the code? I’m a bit stuck for ideas so a fresh pair of eyes on this would be really helpful.

Thanks :)

**Comment by w-k-jones** [comment]
I've been able to reproduce this, and it definitely looks like a bug! Not sure what's going on here but we should look into this fairly urgently. I'll convert this to an issue so we can keep track of it

**Reply by harrietgilmour** [reply]
Thanks, @w-k-jones !

**Comment by w-k-jones** [comment]
Hi @harrietgilmour , I've created a potential fix for this in #316. Would it be possible for you to test this on your data and see if it resolves the problem?

**Reply by harrietgilmour** [reply]
No problem! - I'll take a look at it this afternoon and get back to you

**Reply by harrietgilmour** [reply]
What would be the best way to test this? Clone your strict_threshold_fix branch?

**Reply by w-k-jones** [reply]
The best way would be to clone the main repository and then check out my branch using:

```
git fetch "https://github.com/w-k-jones/tobac.git" strict_threshold_fix
git checkout FETCH_HEAD
```

**Comment by harrietgilmour** [comment]
Hi @w-k-jones , I've tested the fix on my data and it seems to working as expected now. Thank you for all your hard work sorting this out! :)
